### ISSUE TITLE
Opening a file under-merge in Magit Status leads to (error "Lisp nesting exceeds ...)

### ISSUE THREAD

**Issue by xuning97** [hlissner/doom-emacs#745]
### Observed behavior

Magit Status open a file under-merge

### Expected behavior

smerge to merge the file

### Steps to reproduce

1. generate some files under under-merge state
2. open it from magic-status
3. report (error "Lisp nesting exceeds ...)

### System information

<details>
<summary>Click to expand</summary>

```
Doom Doctor
Emacs v27.0.50
Doom v2.0.9 (ee917486)
shell: /bin/bash
Compiled with:
  XPM JPEG TIFF GIF PNG SOUND NOTIFY LIBSELINUX GNUTLS ZLIB OLDXMENU X11 THREADS
uname -a:
  Linux bigisland 3.10.0-862.3.3.el7.x86_64 #1 SMP Fri Jun 15 04:15:27 UTC 2018
  x86_64 x86_64 x86_64 GNU/Linux

Checking your OS...
Checking your fonts...
Checking gnutls/openssl...
Testing your root certificates...
Checking for GNU/BSD tar...
Checking your enabled modules...


Everything seems fine, happy Emacs’ing!
```
</details>

Debugger entered--Lisp error: (error "Lisp nesting exceeds ‘max-lisp-eval-depth’")
  (or doom-inhibit-switch-window-hooks (null window) (eq window (selected-window)) (window-minibuffer-p) (window-minibuffer-p window))
  (if (or doom-inhibit-switch-window-hooks (null window) (eq window (
  doom*switch-window-hooks(#<subr select-window> #<window 214 on  *LV*> norecord)
  apply(doom*switch-window-hooks #<subr select-window> (#<window 214 on  *LV*> norecord))
  #f(advice-wrapper :around #<subr select-window> doom*switch-window-hooks)(#<window 214 on  *LV*> norecord)
  ad-Advice-select-window(#f(advice-wrapper :around #<subr select-window> doom*switch-window-hooks) #<window 214 on  *LV*> norecord)
  apply(ad-Advice-select-window #f(advice-wrapper :around #<subr select-window> doom*switch-window-hooks) (#<window 214 on  *LV*> norecord))
  select-window(#<window 214 on  *LV*> norecord)
  internal--after-save-selected-window((#<window 214 on  *LV*> (#<frame F1 0xc272b0> . #<window 214 on  *LV*>)))
  #f(compiled-function () #<bytecode 0x13810c5>)()
  walk-windows(#f(compiled-function (window) #<bytecode 0x1381095>) 1 t)
  window--process-window-list()
  window--adjust-process-windows()
  #<subr set-window-buffer>(nil #<buffer  *LV*> nil)
  ad-Advice-set-window-buffer(#<subr set-window-buffer> nil #<buffer  *LV*>)
  apply(ad-Advice-set-window-buffer #<subr set-window-buffer> (nil #<buffer  *LV*>))
  set-window-buffer(nil #<buffer  *LV*>)
  #f(compiled-function (buffer-or-name &optional norecord force-same-window) "Display buffer BUFFER-OR-NAME in the selected window.\n\nWARNING: This is NOT the way to work on an$
  apply(#f(compiled-function (buffer-or-name &optional norecord force-same-window) "Display buffer BUFFER-OR-NAME in the selected window.\n\nWARNING: This is NOT the way to work$
  (prog1 (apply orig-fn buffer-or-name args) (save-current-buffer (set-buffer buffer-or-name) (run-hooks '
  (let ((doom-inhibit-switch-buffer-hooks t)) (run-hooks 'doom-exit-buffer-hook) (prog1 (apply orig-fn buffer-or-name args) (save-current-buffer (
  (if (or doom-inhibit-switch-buffer-hooks (eq (
  doom*switch-buffer-hooks(#f(compiled-function (buffer-or-name &optional norecord force-same-window) "Display buffer BUFFER-OR-NAME in the selected window.\n\nWARNING: This is $
  apply(doom*switch-buffer-hooks #f(compiled-function (buffer-or-name &optional norecord force-same-window) "Display buffer BUFFER-OR-NAME in the selected window.\n\nWARNING: Th$
  #f(advice-wrapper :around #f(compiled-function (buffer-or-name &optional norecord force-same-window) "Display buffer BUFFER-OR-NAME in the selected window.\n\nWARNING: This is$
  ad-Advice-switch-to-buffer(#f(advice-wrapper :around #f(compiled-function
  apply(ad-Advice-switch-to-buffer #f(advice-wrapper :around #f(compiled-function
  switch-to-buffer(" *LV*")
  lv-window()
  lv-message(#("                                                         [smerge]\n  Movement   Keep           Diff              Other\n  ╭──────────────────────────────────────$
  hydra-show-hint((concat (format "                                                         [smerge]\n  Movement   Keep           Diff              Other\n  ╭───────────────────$
  +hydra-smerge/body()
  run-hooks(smerge-mode-hook smerge-mode-on-hook)
  smerge-mode(1)
  (if (not smerge-mode) (smerge-mode 1))
  +hydra-smerge/body()
  *Backtrace*  1:30 Top                                                                                                                                           UTF-8  Debugger
C-x 1

**Comment by hlissner** [comment]
This should be resolved as of 29c2632. 

`+hydra-smerge/body` was in `smerge-mode-hook`. When the hydra is activated, it tries to activate smerge-mode again. When the hydra is closed, the hook runs again, re-enabling smerge-mode. Somewhere in this cyclical chain of events is an infinite recursion, so I've removed `+hydra-smerge/body` from `smerge-mode-hook`.

Let me know if the issue isn't resolved!

**Comment by xuning97** [comment]
Thanks
